reference/src/glossary.md

```diff
@@ -53,12 +53,72 @@ For some more information, see [this blog post](https://www.ralfj.de/blog/2018/0
 
 #### Layout
 
-The *layout* of a type defines its size and alignment as well as the offsets of its subobjects (e.g. fields of structs/unions/enum/... or elements of arrays).
-Moreover, the layout of a type records its *function call ABI* (or just *ABI* for short): how the type is passed *by value* across a function boundary.
+The *layout* of a type defines:
+
+* its size,
+* its alignment,
+* its field offsets (e.g. fields of structs, union, enums, arrays, ...)
+* its [niches][Niche],
+* its call abi.
 
 Note: Originally, *layout* and *representation* were treated as synonyms, and Rust language features like the `#[repr]` attribute reflect this. 
 In this document, *layout* and *representation* are not synonyms.
 
+Note: due to discriminant-elision optimizations, niches are required to compute
+the layout of, e.g., `Option<T>`, from the layout of `T`.
+
+#### Call ABI
+
+The *call ABI* determines how a type is passed *by value* across a function boundary.
+
+Note: The call ABI is not stable. Currently, it is one of:
+
+```rust,norun
+pub enum Abi {
+    Uninhabited,
+    Scalar(Scalar),
+    Vector {
+        element: Scalar,
+        count: u64
+    },
+    Aggregate {
+        // If true, the layout size of the aggregate is exact.
+        // Otherwise, the layout size of the aggregate is
+        // a lower bound.
+        sized: bool,
+    }
+}
+```
+
+where
+
+```rust,norun
+enum Integer { I8, I16, I32, I64, I128, }
+enum FloatTy { F32, F64, }
+
+enum Primitive {
+    Int(Integer, /*signedness:*/ bool),
+    Float(FloatTy),
+    Pointer
+}
+
+struct Scalar {
+    value: Primitive,
+    valid_range: RangeInclusive<u128>,
+}
+```
+
+For example, right now, the call ABI of:
+
+* `i32` is `Scalar`,
+* `struct Wrapper(i32);` (without `repr(transparent)`) is also `Scalar` (only larger `struct`s are aggregates),
```

Comment:
Given that you mention `repr(Rust)` below, I'd also write `#[repr(Rust)] struct Wrapper` here. Otherwise people have to know that this is the default to even make the connection.